src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -149,7 +149,8 @@ public Task<RestoreResult> ExecuteAsync()
         {
             if (_request.Project.TargetFrameworks.Count == 0)
             {
-                _log.LogError(Strings.FormatLog_ProjectDoesNotSpecifyTargetFrameworks(_request.Project.Name));
+                var jsonFileName = !String.IsNullOrEmpty(_request.Project.FilePath) ? _request.Project.FilePath.Split('\\').Last() : "project.json";
```

Comment:
Use Path.GetFileName here, splitting on \ will not work on mono and RestoreCommand is xplat.